src/main/java/bridge/Controller.java

```diff
@@ -0,0 +1,71 @@
+package bridge;
+
+import bridge.domains.BridgeGame;
+import bridge.domains.BridgeMaker;
+import bridge.domains.BridgeMap;
+import bridge.domains.BridgeRandomNumberGenerator;
+import bridge.dto.GameInformationDto;
+import bridge.services.BridgeService;
+import bridge.views.InputView;
+import bridge.views.OutputView;
+
+import java.util.List;
+
+public final class Controller {
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final BridgeService bridgeService;
+
+    public Controller() {
+        inputView = new InputView();
+        outputView = new OutputView();
+
+        BridgeMap bridgeMap = makeBridge();
+        bridgeService = new BridgeService(bridgeMap, new BridgeGame(bridgeMap));
+        play();
+        finishGame(bridgeService.checkBridge());
+    }
+
+    private BridgeMap makeBridge() {
+        BridgeMap bridgeMap;
+        int readSize = inputView.readBridgeSize();
+
+        try {
+            bridgeMap = new BridgeMap(new BridgeMaker(new BridgeRandomNumberGenerator()), readSize);
+        } catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+            bridgeMap = makeBridge();
+        }
+        return bridgeMap;
+    }
```

Comment:
```suggestion
private BridgeMap makeBridge() {
        int readSize = inputView.readBridgeSize();

        try {
            return new BridgeMap(new BridgeMaker(new BridgeRandomNumberGenerator()), readSize);
        } catch (IllegalArgumentException e) {
            System.out.println(e.getMessage());
            return makeBridge();
        }
    }
```
1. 재귀적으로 재입력 받기 위함이라면 위와 같이 쓸 수 있을 것 같네요.
2. System.out.println은 출력문입니다. OutputView의 역할이 아닐까요? 컨트롤러가 직접 UI 동작을 하면 어떤 문제가 있을까요?

Reply:
2.
OutputView가 있음에도 controller가 view의 역할을 수행하면 
controller가 2가지 역할을 수행하는 것이므로 mvc 패턴의 의미가 사라집니다.